rllib/evaluation/rollout_worker.py

```diff
@@ -267,133 +286,183 @@ def __init__(
+            )
+        if model_config != DEPRECATED_VALUE:
+            deprecation_warning("model_config", "config.training(model=..)", error=True)
+        if remote_worker_envs != DEPRECATED_VALUE:
+            deprecation_warning(
+                "remote_worker_envs",
+                "config.rollouts(remote_worker_envs=..)",
+                error=True,
+            )
+        if remote_env_batch_wait_ms != DEPRECATED_VALUE:
+            deprecation_warning(
+                "remote_env_batch_wait_ms",
+                "config.rollouts(remote_env_batch_wait_ms=..)",
+                error=True,
+            )
+        if soft_horizon != DEPRECATED_VALUE:
+            deprecation_warning(
+                "soft_horizon", "config.rollouts(soft_horizon=..)", error=True
+            )
+        if no_done_at_end != DEPRECATED_VALUE:
+            deprecation_warning(
+                "no_done_at_end", "config.rollouts(no_done_at_end=..)", error=True
+            )
+        if fake_sampler != DEPRECATED_VALUE:
+            deprecation_warning(
+                "fake_sampler", "config.rollouts(fake_sampler=..)", error=True
+            )
+        if seed != DEPRECATED_VALUE:
+            deprecation_warning("seed", "config.debugging(seed=..)", error=True)
+        if log_level != DEPRECATED_VALUE:
+            deprecation_warning(
+                "log_level", "config.debugging(log_level=..)", error=True
+            )
+        if callbacks != DEPRECATED_VALUE:
+            deprecation_warning(
+                "callbacks", "config.callbacks([DefaultCallbacks subclass])", error=True
+            )
+        if disable_env_checking != DEPRECATED_VALUE:
+            deprecation_warning(
+                "disable_env_checking",
+                "config.environment(disable_env_checking=..)",
+                error=True,
+            )
+        if policy_mapping_fn != DEPRECATED_VALUE:
+            deprecation_warning(
+                "policy_mapping_fn",
+                "config.multi_agent(policy_mapping_fn=..)",
+                error=True,
+            )
+        if policies_to_train != DEPRECATED_VALUE:
+            deprecation_warning(
+                "policies_to_train",
+                "config.multi_agent(policies_to_train=..)",
+                error=True,
+            )
+        if extra_python_environs != DEPRECATED_VALUE:
+            deprecation_warning(
+                "extra_python_environs",
+                "config.python_environment(extra_python_environs_for_driver=.., "
+                "extra_python_environs_for_worker=..)",
+                error=True,
+            )
 
         self._original_kwargs: dict = locals().copy()
         del self._original_kwargs["self"]
 
         global _global_worker
         _global_worker = self
 
-        # set extra environs first
-        if extra_python_environs:
-            for key, value in extra_python_environs.items():
+        if config is None or isinstance(config, dict):
```

Comment:
Similar to the logic in `Algorithm`: Make sure, we are working with an AlgorithmConfig object under the hood, then freeze.